### PR TITLE
Updates for Cromwell 24

### PR DESCRIPTION
**Pull request by jmthibault79**
GAWB-1464

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala [hunk 1]
```diff
@@ -266,7 +267,7 @@ class RemoteServicesMockServer(port:Int) {
       request()
         .withMethod("POST")
         .withPath(submissionBatchPath)
-        .withBody(new ParameterBody(new Parameter("workflowOptions", "two_second_delay")))
```
Comment (jmthibault79) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/mock/RemoteServicesMockServer.scala:269]: `workflowOptions` are now passed as Multipart form data

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala [hunk 2]
```diff
@@ -675,9 +675,11 @@ class MethodConfigApiServiceSpec extends ApiServiceSpec {
       sealRoute(services.methodConfigRoutes) ~>
       check {
         assertResult(StatusCodes.OK) { status }
-        assertResult(MethodInputsOutputs(Seq(MethodInput("three_step.cgrep.pattern","String",false)),Seq(MethodOutput("three_step.ps.procs","File"), MethodOutput("three_step.cgrep.count","Int"), MethodOutput("three_step.wc.count","Int")))) {
-          responseAs[MethodInputsOutputs]
-        }
+        val expectedIn = Seq(MethodInput("three_step.cgrep.pattern","String",false))
+        val expectedOut = Seq(MethodOutput("three_step.ps.procs","File"), MethodOutput("three_step.cgrep.count","Int"), MethodOutput("three_step.wc.count","Int"))
+        val result = responseAs[MethodInputsOutputs]
+        assertSameElements(expectedIn, result.inputs)
```
Comment (jmthibault79) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala:681]: Ordering changed, but order is not important

File: project/Dependencies.scala [hunk 3]
```diff
@@ -27,7 +27,7 @@ object Dependencies {
     "io.spray" %% "spray-client" % sprayV,
     "org.webjars" % "swagger-ui" % "2.1.1",
     "org.apache.commons" % "commons-jexl" % "2.1.1",
-    "org.broadinstitute" %% "wdl4s" % "0.5" exclude("org.scalaz", "scalaz-core_2.11") exclude("io.spray", "spray-json_2.11"),
+    "org.broadinstitute" %% "wdl4s" % "0.9-40dddbc-SNAP" exclude("org.scalaz", "scalaz-core_2.11") exclude("io.spray", "spray-json_2.11"),
```
Comment (jmthibault79) [inline project/Dependencies.scala]: Will update to a WDL release, probably 0.10

**Review by dvoet — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala [hunk 4]
```diff
@@ -27,8 +27,8 @@ class HttpExecutionServiceDAO( executionServiceURL: String, submissionTimeout: F
     import system.dispatcher
 
     val pipeline = addAuthHeader(userInfo) ~> sendReceive ~> unmarshal[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]]
-    val formData = FormData(Seq("wdlSource" -> wdl, "workflowInputs" -> inputs.mkString("[", ",", "]")) ++ options.map("workflowOptions" -> _).toSeq)
-    pipeline(Post(url,formData))
+    val formData = Map("wdlSource" -> BodyPart(wdl), "workflowInputs" -> BodyPart(inputs.mkString("[", ",", "]"))) ++ options.map("workflowOptions" -> BodyPart(_))
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala:30]: does there need to be a change to `ExecutionServiceWorkflowOptions` to support call caching, or is it an `Optional`?

Comment (jmthibault79) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala:30]: This PR does not address call caching support

Comment (helgridly) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala:30]: My question is whether the workflow options struct has changed to support call caching in C24, and whether our case class needs to change in accordance with that.

Comment (jmthibault79) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala:30]: Oh I see.  No, the call does not require a call caching entry.

**Review by ansingh7115 — APPROVED** [review]

**Review by helgridly — APPROVED** [review]

**Pull request merged**